model/lifecycle.go

```diff
@@ -1314,6 +1296,58 @@ func createOneTask(id string, buildVarTask BuildVariantTaskUnit, project *Projec
 	return t, nil
 }
 
+func generateDistros(id string, buildVarTask BuildVariantTaskUnit, buildVariant *BuildVariant, project *Project, v *Version) (string, []string) {
+	var (
+		distroID      string
+		distroAliases []string
+	)
+
+	if len(buildVarTask.RunOn) > 0 {
```

Comment:
This if-elseif-else might be more easily readable if it went something like:
```go
if len(buildVarTask.RunOn) > 0 {
    // Set distro ID and aliases...
    return distroID, distroAliases
else if len(buildVariant.RunOn) > 0 {
    // Set distro ID and aliases...
    return distroID, distroAliases
}

return errors.New("task does not have a distro to run on")
```
It would also remove the need to declare `distroID` and `distroAliases` at the top of the function.